tests/staking/apy.ts

```diff
@@ -0,0 +1,277 @@
+import * as anchor from "@project-serum/anchor";
+import { findProgramAddressSync } from "@project-serum/anchor/dist/cjs/utils/pubkey";
+import { Token, TOKEN_PROGRAM_ID, u64 } from "@solana/spl-token";
+import { BN } from "bn.js";
+import { Staking } from "../../target/types/staking";
+import assert from "assert";
+import { AnchorError } from "@project-serum/anchor";
+import { sleep } from "@project-serum/common";
+import { getClock } from "../clock";
+
+const { PublicKey, Keypair, LAMPORTS_PER_SOL } = anchor.web3;
+type BN = anchor.BN;
+type PublicKey = anchor.web3.PublicKey;
+type Keypair = anchor.web3.Keypair;
+
+const program = anchor.workspace.Staking as anchor.Program<Staking>;
+const provider = anchor.AnchorProvider.env();
+anchor.setProvider(provider);
+
+const adminKeypair = new Keypair();
+const userKeypair = new Keypair();
+
+async function waitUntilOnChainTime(
+  connection: anchor.web3.Connection,
+  time: number
+) {
+  let onChainTime = 0;
+  do {
+    const clock = await getClock(connection);
+    onChainTime = clock.info.unixTimestamp;
+  } while (onChainTime <= time);
+}
+
+async function computeApy(program: anchor.Program<Staking>, pool: PublicKey) {
+  const poolState = await program.account.pool.fetch(pool);
+  const clock = await getClock(program.provider.connection);
+  const lockedToken = poolState.totalStaked.toNumber();
+  if (lockedToken == 0) {
+    return 0;
+  }
+  // Farm expired
+  if (poolState.xmerRewardEndTimestamp.toNumber() < clock.info.unixTimestamp) {
+    return 0;
+  }
+  const rewardPerDay = poolState.xmerRewardRate.div(new BN(365));
+  const rewardPerTokenPerDay = rewardPerDay.toNumber() / lockedToken;
+  return (1 + rewardPerTokenPerDay) ** 365 - 1 * 100;
```

Comment:
Should be ((1 + rewardPerTokenPerDay) ** 365 - 1) * 100 ?

Reply:
Yes, fixed it.